build-system/tasks/extension-generator/extension-doc.template.md

```diff
@@ -0,0 +1,220 @@
+  * Do not include markdown formatting in the frontmatter - plain text and punctionation only!
+  * Remove this comment!
+-->
+
+<!--
+Copyright ${year} The AMP HTML Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS-IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+# \${name}
+
+<!--
+  If the component is relevant for more than one format and operates differently between these
+  formats, include and filter multiple content blocks and code samples.
+-->
+
+## Usage
+
+One to three paragraphs explaining the component usage. List important functionality. Explain why developers care about it.
+
+[filter formats=“websites”]
+
+Below is an example for websites.
+
+[example preview="inline" playground="true" imports="${name}"]
+
+```html
+<${name} required-attribute>
+  I am a hello world inline executable code sample for websites!
+</${name}>
+```
+
+[/example][/filter]
+
+<!--
+  * [Read more about filtering sections](https://amp.dev/documentation/guides-and-tutorials/contribute/contribute-documentation/formatting/?format=websites#filtering-sections)
+  * [Read more about executable code samples](https://amp.dev/documentation/guides-and-tutorials/contribute/contribute-documentation/formatting/?format=websites#preview-code-samples)
+ -->
+
+[filter formats=“ads”]
+
+Below is an example for ads.
+
+[example preview=“inline” playground=“true” imports="${name}"]
+
+```html
+<${name} required-attribute>
+  I am a hello world inline executable code sample for ads!
+</${name}>
+```
+
+[/example][/filter]
+
+### Standalone use outside valid AMP documents (optional)
+
+Bento AMP allows you to use AMP components in non-AMP pages without needing to commit to fully valid AMP. You can take these components and place them in implementations with frameworks and CMSs that don't support AMP. Read more in our guide [Use AMP components in non-AMP pages](https://amp.dev/documentation/guides-and-tutorials/start/bento_guide/)
+
+#### Example
+
+Code snippet of Bento component in standalone usage.
+
+#### Interactivity and API usage
+
+Bento enabled components in standalone use are highly interactive through their API. Because standalone mode doesn't require valid AMP, you can create custom interactivity without the use of [`amp-bind`](https://amp.dev/documentation/components/amp-bind/?format=websites) implementations and the API replaces [AMP actions and events](https://amp.dev/documentation/guides-and-tutorials/learn/amp-actions-and-events/?format=websites).
```

Comment:
Nit: I wonder if we can rephrase this a bit more direct? E.g. "In Bento the element's API replaces AMP Actions and events and amp-bind. Instead you can listen to events and call API directly." Or something like this?